pkg/module/go_get_fetcher.go

```diff
@@ -0,0 +1,159 @@
+package module
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+var (
+	// ErrLimitExceeded signals that github.com refused to serve the request due to exceeded quota
+	ErrLimitExceeded = errors.New("github limit exceeded")
+)
+
+type goGetFetcher struct {
+	fs           afero.Fs
+	repoURI      string
+	version      string
+	goBinaryName string
+}
+
+// NewGoGetFetcher creates fetcher which uses go get tool to fetch modules
+func NewGoGetFetcher(goBinaryName string, fs afero.Fs, repoURI, version string) (Fetcher, error) {
+	if repoURI == "" {
+		return nil, errors.New("invalid repository identifier")
+	}
+
+	return &goGetFetcher{
+		fs:           fs,
+		repoURI:      repoURI,
+		version:      version,
+		goBinaryName: goBinaryName,
+	}, nil
+}
+
+// Fetch downloads the sources and returns path where it can be found. Fetch will always
+// return a non-nil Ref.
+//
+// TODO: If an error is returned, the Fetch method will likely fail, but you should always call
+// Close on the returned
+func (g *goGetFetcher) Fetch(mod, ver string) (Ref, error) {
+	// setup the GOPATH
+	goPathRoot, err := afero.TempDir(g.fs, "", "athens")
+	if err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+	sourcePath := filepath.Join(goPathRoot, "src")
+	modPath := filepath.Join(sourcePath, getRepoDirName(g.repoURI, g.version))
+	if err := g.fs.MkdirAll(modPath, os.ModeDir|os.ModePerm); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+
+	// setup the module with barebones stuff
+	if err := prepareStructure(g.fs, modPath); err != nil {
+		// TODO: return a ref for cleaning up the goPathRoot
+		return nil, err
+	}
+
+	cachePath, err := getSources(g.goBinaryName, g.fs, goPathRoot, modPath, mod, ver)
+	if err != nil {
```

Comment:
you can simplify all of this to just
```go
return newDiskRef(g.fs, cachePath, ver), err
```
the error will be nil or not and you want to return the ref either way.